R/assignment_linter.R

```diff
@@ -95,16 +112,16 @@ assignment_linter <- function(allow_cascading_assign = TRUE, allow_right_assign
     }
 
     operator <- xml2::xml_text(bad_expr)
-    lint_message_fmt <- ifelse(
-      operator %in% c("<<-", "->>"),
-      "%s can have hard-to-predict behavior; prefer assigning to a specific environment instead (with assign() or <-).",
-      "Use <-, not %s, for assignment."
-    )
+    lint_message_fmt <- "Use <-, not %s, for assignment."
+    lint_message_fmt[operator %in% c("<<-", "->>")] <-
+      "%s can have hard-to-predict behavior; prefer assigning to a specific environment instead (with assign() or <-)."
+    lint_message_fmt[operator == "%<>%"] <-
+      "Avoid the assignment pipe %s; prefer using <- and %%>%% separately."
```

Comment:
I use this operator all the time, and I am curious to understand _why_ it should be avoided. Currently, the message only says one should avoid it, without giving any further rationale. 

I am assuming it's because this reduces readability? Is it because it's not idiomatic of R, unlike other languages where "in-place" modifying operators are common (e.g. `x += 1;`, instead of `x = x + 1;`)?

Reply:
I think that's the gist of it. it also gets confusing to see it chained.

I was drawing a blank myself explaining it concisely, and the style guide is also very vague.

Reply:
Filed https://github.com/tidyverse/style/issues/203.

Let's leave the wording vague for now, pending clarification in the guide?

Reply:
SGTM.